### DISCUSSION TITLE
Sending information back and forth between Unity and React

### DISCUSSION THREAD

**Discussion by flamypow** [jeffreylanters/react-unity-webgl#184] [Question and Answer] [answered]
so I am trying to send some information into Unity when Unity sends something to me. 
If I go 

```js
unityContext.on("Event", (something) => {
  SendFunction();
}
async function SendFunction(){
  unityContext.send("GameController", "RecieveInfo", "here is a string I want to send to Unity");
}
```

the string seems to be not received by Unity. Assuming Unity side is built properly and if I just go SendFunction(); outside of unityContext.on it works. 
What is the proper way of sending things in situation like this? I think I am doing something wrong here but I don't know what. 
edit: found the problem. Very interesting how if I put the constructor of unitycontext inside of the page a lot of functions still work except this one.

**Comment by ejustin-wormholelabs** [comment, marked as answer]
Could it be a typo ... "RecieveInfo" should be "ReceiveInfo"?

**Reply by flamypow** [reply]
Yeah It was something I missed, misplaced where I made the unitycontext oops.
